### PR TITLE
feat: [FC-5] custom bubble message

### PR DESCRIPTION
**Pull request by huydosgtech**

### CODE REVIEW

File: example/android/app/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -1,7 +1,12 @@
 <manifest xmlns:android="http://schemas.android.com/apk/res/android">
 
     <uses-permission android:name="android.permission.INTERNET" />
+    <uses-permission android:name="android.permission.CAMERA" />
+    <uses-permission android:name="android.permission.READ_EXTERNAL_STORAGE" />
+    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE" />
```
Comment (loc-nguyenthien) [inline example/android/app/src/main/AndroidManifest.xml:6]: We will need `<uses-permission android:name="android.permission.READ_MEDIA_IMAGES"/>` for android 13

File: src/chat/ChatScreen.tsx [hunk 2]
```diff
@@ -154,6 +175,109 @@ export const ChatScreen: React.FC<ChatScreenProps> = ({
...
+        createdAt: Date.now(),
+        status: MessageStatus.sent,
+      };
+
+      await firebaseInstance.sendMessageWithFile(message);
+      const imgVideoMessage = {
+        ...message,
+        createdAt: message.createdAt || Date.now(),
+        _id: `${Math.floor(Math.random() * 1000000)}`,
+        id: `${Math.floor(Math.random() * 1000000)}`,
```
[the remark this answers is not in the document]
Comment (huydosgtech) [reply src/chat/ChatScreen.tsx]: yes, I use uuid for generated ID

File: src/chat/components/SelectedImage.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+          }}
+          resizeMode={FastImage.resizeMode.contain}
+        />
+      </TouchableOpacity>
+    </Modal>
+  );
+};
+
+const styles = StyleSheet.create({
+  loadingContainer: {
```
[the remark this answers is not in the document]
Comment (huydosgtech) [reply src/chat/components/SelectedImage.tsx]: removed loadingContainer

File: example/android/app/src/main/AndroidManifest.xml (hunk 1 above)
Comment (huydosgtech) [reply example/android/app/src/main/AndroidManifest.xml:6]: yes, I added new

File: src/chat/components/CustomBubble.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+    >
+      <Pressable
+        onPress={() =>
+          message.path &&
+          message.type === 'photo' &&
+          selectedImgVideoUrl(message.path)
+        }
+        style={styles.bubble}
+      >
+        {message.type === 'photo' ? (
```
Comment (baonguyenhsts) [inline src/chat/components/CustomBubble.tsx:42]: The type of message is photo, video and text. Where is the logic for "text"?

**Review by baonguyenhsts — CHANGES_REQUESTED** [review]

File: src/chat/components/CustomBubble.tsx (hunk 4 above)
Comment (huydosgtech) [reply src/chat/components/CustomBubble.tsx]: the text logic used default Bubble. I already changed name to **CustomImageVideoBubble** more easy to read

File: src/chat/components/CustomImageVideoBubble.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+  TouchableOpacity,
+} from 'react-native';
+import Video, { VideoRef } from 'react-native-video';
+import FastImage from 'react-native-fast-image';
+import { MessageTypes, type MessageProps } from '../../interfaces';
+
+interface CustomImageVideoBubbleProps {
+  message: MessageProps;
+  position: 'left' | 'right';
+  selectedImgVideoUrl: (url: string) => void;
```
Comment (danhmaisgt) [inline src/chat/components/CustomImageVideoBubble.tsx:16]: I think this prop name should begin with a verb, so it better be `onPress` or `onSelectCallback`

File: src/chat/components/CustomImageVideoBubble.tsx [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+                onEnd={() => {
+                  setIsPauseVideo(true);
+                  videoRefs.current?.seek(0);
+                }}
+                ref={videoRefs}
+              />
+              {isPauseVideo && (
+                <Image
+                  style={styles.playIcon}
+                  source={require('../../images/play.png')}
```
Comment (danhmaisgt) [inline src/chat/components/CustomImageVideoBubble.tsx:71]: Can you support to customize this icon?

File: src/chat/components/CustomImageVideoBubble.tsx [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+    >
+      <Pressable
+        onPress={() =>
+          message.path &&
+          message.type === MessageTypes.image &&
+          selectedImgVideoUrl(message.path)
+        }
+        style={styles.bubble}
+      >
+        {message.type === MessageTypes.image ? (
```
Comment (honghoangsts) [inline src/chat/components/CustomImageVideoBubble.tsx:42]: please create a function for each type and add props to custom style and the render of each type

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: src/chat/components/CustomImageVideoBubble.tsx (hunk 6 above)
Comment (huydosgtech) [reply src/chat/components/CustomImageVideoBubble.tsx]: ok, I added support customize icon

File: src/chat/components/CustomImageVideoBubble.tsx (hunk 7 above)
Comment (huydosgtech) [reply src/chat/components/CustomImageVideoBubble.tsx]: ok. I already separate to small function and add custom style for it

**Pull request merged**